src/ActivityHeartbeatingCancellation/MyWorkflow.workflow.cs

```diff
@@ -0,0 +1,30 @@
+using Microsoft.Extensions.Logging;
+using Temporalio.Exceptions;
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.ActivityHeartbeatingCancellation;
+
+[Workflow]
+public class MyWorkflow
+{
+    [WorkflowRun]
+    public async Task RunAsync()
+    {
+        try
+        {
+            await Workflow.ExecuteActivityAsync(
+                () => MyActivities.FakeProgressAsync(1000),
+                new()
+                {
+                    StartToCloseTimeout = TimeSpan.FromMinutes(5),
+                    HeartbeatTimeout = TimeSpan.FromSeconds(3),
+                    // Don't send rejection to our Workflow until the Activity has confirmed cancellation
+                    CancellationType = ActivityCancellationType.WaitCancellationCompleted,
+                });
+        }
+        catch (ActivityFailureException e) when (e.InnerException is CancelledFailureException)
+        {
```

Comment:
```suggestion
        catch (TemporalException e) when (TemporalException.IsCancelledException(e))
        {
            // This catches the cancel just for demonstration, you usually don't want to catch it
```

I think this is a better sample showing how to catch cancel generally and also adds a comment making it clear that usually you don't want to catch it